### PR TITLE
Trim away netframework targets in source-build

### LINKED ISSUE #74950: TFM trimming, and elimination of net4* targeting in Linux source-build
### Description

Trim away unneeded TFMs, and conditionalize away non-applicable TFMs for Linux source build.

**Requires** https://github.com/dotnet/source-build/issues/3013. source-build net4* targeting elimination required https://github.com/dotnet/source-build/issues/2974

The purpose of this operation is twofold:
- [ ] Reduce the number of TFMs present overall in .NET's build, potentially improving build time.
- [ ] Eliminate framework TFMs, drastically reducing the size of the source build repository size.

When TFMs other than the latest .NET Core TFM are targeted in source-build, the reference assets for that TFM must be provided via a source build reference package. This is a 'text only' package that is assembled from IL at the beginning of the build to satisfy the required reference assets. In Microsoft's build, these would be restored from the internet, but in source build this is not allowed.

These SBRPs are large. All told, the total size contribution for just the netframework 4* TFMs is around 3GB. Not only does this have a significant contribution to the source build repo size, but it also affects build time. For Linux source build, we spend large amounts of time building assets that are simply thrown away at the end.

An 'ideal' would be a single TFM used throughout all of the .NET build, source build or otherwise. This is not possible. There are legitimate reasons to target older TFMs, especially netstandard. However, we should be able to justify each additional TFM.

To complete this work, repositories will use the tooling provided via https://github.com/dotnet/source-build/issues/2901 to identify the TFMs in use in their repository. For each TFM, they should do the following:
- [ ] If the TFM can be eliminated, or upgraded to latest (e.g. target the latest netcore explicitly, or use net472), do so. netframework TFM removal for tools and internal code is likely low hanging fruit.
- [ ] **Requires https://github.com/dotnet/source-build/issues/2974** If the TFM is a framework TFM, it should be conditionalized out in source build for Linux.

It is a goal that the form of this conditionalization remain consistent across projects, and used relatively sparingly. How this conditionalization should be done will be designed in https://github.com/dotnet/source-build/issues/2974. Once that is done, this issue will be updated with implementation details.

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
﻿### Summary of the changes

This is part of the source-build work to enable trimming of net4* targets. Parent repos (`installer` and `sdk`) have been updated, so the next step is to update dependencies, like `runtime`.

Here's how this is consumed: https://github.com/dotnet/arcade/pull/12785

Top-level tracking issue: https://github.com/dotnet/source-build/issues/3014

This essentially just avoids building net4* TFMs when building Linux source build.

Fixes: https://github.com/dotnet/runtime/issues/74950

### CODE REVIEW

**Review by mmitche — APPROVED** [review]

**Comment by mmitche** [comment]
VERY cool.

**Review by crummel — APPROVED** [review]

File: eng/SourceBuild.props [hunk 1]
```diff
@@ -43,6 +43,8 @@
       <InnerBuildArgs Condition="'$(ContinuousIntegrationBuild)' != ''">$(InnerBuildArgs) /p:ContinuousIntegrationBuild=$(ContinuousIntegrationBuild)</InnerBuildArgs>
       <InnerBuildArgs>$(InnerBuildArgs) /p:AdditionalRuntimeIdentifierParent=$(BaseOS)</InnerBuildArgs>
       <InnerBuildArgs Condition="'$(SourceBuildUseMonoRuntime)' == 'true'">$(InnerBuildArgs) /p:PrimaryRuntimeFlavor=Mono /p:RuntimeFlavor=Mono</InnerBuildArgs>
+      <!-- Unset NetFrameworkMinimum property, to disable netfx projects. -->
+      <InnerBuildArgs>$(InnerBuildArgs) /p:NetFrameworkMinimum=</InnerBuildArgs>
```
Comment (ViktorHofer) [inline eng/SourceBuild.props:47]: This should be done in Directory.Build.props instead in order for it work with the -sb switch.

**Review by ViktorHofer — CHANGES_REQUESTED** [review]

File: eng/SourceBuild.props (hunk 1 above)
Comment (NikolaMilosavljevic) [reply eng/SourceBuild.props:47]: Thanks - will fix this.

Comment (NikolaMilosavljevic) [reply eng/SourceBuild.props]: Fixed with commit: https://github.com/dotnet/runtime/pull/83899/commits/76779f9141352e0fd2e220ac747165584747f547

**Review by ViktorHofer — APPROVED** [review]
Thanks, LGTM

**Pull request merged**